modules/contact-form/admin.php

```diff
@@ -902,3 +940,65 @@ function grunion_recheck_queue() {
 }
 
 add_action( 'wp_ajax_grunion_recheck_queue', 'grunion_recheck_queue' );
+
+/**
+ * Delete a number of spam feedbacks via an AJAX request.
+ */
+function grunion_delete_spam_feedbacks() {
+	if ( ! wp_verify_nonce( $_POST['nonce'], 'jetpack_delete_spam_feedbacks' ) ) {
+		wp_send_json(
+			array(
+				'error' => __( 'You don&#8217;t have permission to do that.', 'jetpack' ),
+			)
+		);
+
+		return;
+	}
+
+	$query = 'post_type=feedback&post_status=spam';
+
+	if ( isset( $_POST['limit'] ) ) {
+		$query .= '&posts_per_page=' . intval( $_POST['limit'] );
+	}
+
+	$spam_feedbacks = get_posts( $query );
```

Comment:
I wonder if we should tighten this a bit, use `WP_Query`, and always provide a limit to avoid having too big queries? Something like this:

```php
$number_to_delete = absint( $_POST['limit'] );
if ( 1 > $number_to_delete ) {
		$number_to_delete  = 20;
} elseif ( 100 < $number_to_delete ) {
	// Let's avoid processing too many at a time.
	$number_to_delete = 100;
}

$query_args     = array(
	'post_type'      => 'feedback',
	'post_status'    => 'spam',
	'posts_per_page' => $number_to_delete,
);
$spam_feedbacks = new WP_Query( $query_args );
```

Reply:
I've updated this in 3696f911e to eliminate the $_POST['limit'] parameter altogether, since it wasn't really necessary.  I've also switched it to use WP_Query and put a limit on the number of feedbacks that can be selected, since the limit is now configurable via a filter.